### ISSUE TITLE
Why use neg_avg_factor = (1 - p_neg_weight).sum() ?

### ISSUE THREAD

**Issue by Icecream-blue-sky** [strongwolf/DW#30]
In loss function:
```python
neg_avg_factor = (1 - p_neg_weight).sum()
...
cls_neg_loss = sum(cls_neg_loss_list) / neg_avg_factor
```
Why not use neg_avg_factor = (p_neg_weight).sum() ?

**Comment by strongwolf** [comment]
The value of (p_neg_weight).sum() is too small and it will cause unstable training.
